src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,169 @@
+import getSponsoredVideos from 'utils/getSponsoredVideos';
+import waitFor from 'utils/waitFor';
+
+interface WindowJWPlayer extends Window {
+	jwplayer?: JWPlayerApi;
+	sponsoredVideos?: string[];
+}
+
+declare let window: WindowJWPlayer;
+
+const JwPlayerWrapperWithStrategyRules: React.FC<JwPlayerWrapperProps> = ({
+	config,
+	onReady,
+	getDismissed = () => false,
+	onComplete,
+	className,
+	stopAutoAdvanceOnExitViewport,
+	shouldLoadSponsoredContentList = true,
+	jwPlayerContainerEmbedId = 'featured-video__player',
+}) => {
+	const { setPlayer, setConfig } = useContext(PlayerContext);
+	const videoIndexRef = React.useRef(0);
+
+	useEffect(() => {
+		if (shouldLoadSponsoredContentList) {
+			const retrieveSponsoredVideo = async () => {
+				if (window?.sponsoredVideos?.length > 0) {
+					console.debug('sponsoredVideos already retrieved');
+					return;
+				}
+
+				const sponsoredVideoResponse = await getSponsoredVideos();
+				if (sponsoredVideoResponse && typeof window !== undefined) {
+					window.sponsoredVideos = sponsoredVideoResponse;
+					console.debug('Retrieved sponsoredVideos list');
+				} else {
+					console.debug(
+						'Could not set sponsored videos. Either window the fetched sponsoredVideo list were undefined.',
+					);
+				}
+			};
+			retrieveSponsoredVideo().catch((e) => {
+				console.error('There was an issue with retrieving Sponsored Videos. ', e);
+			});
+		} else {
+			console.debug('Loading of Sponsored Content Video List was disabled.');
+		}
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_INIT_RENDER);
+		initPlayer();
+	}, []);
+
+	const registerEventHandlers = (playerInstance) => {
+		playerInstance.on(JWEvents.AD_PAUSE, ({ pauseReason, viewable }: JWPauseEvent) => {
+			// Keep playing the ad when the user closed the mini player
+			if (viewable === 0 && pauseReason === 'external') {
+				playerInstance.play();
+			}
+		});
+
+		playerInstance.on(JWEvents.PLAY, ({ playReason, viewable }: JWPlayEvent) => {
+			const dismissed = getDismissed();
+			// Pause the content play when the user closed the mini player playing the ad
+			if (dismissed && viewable === 0 && playReason === 'autostart') {
+				playerInstance.pause();
+			}
+		});
+
+		playerInstance.on(JWEvents.READY, () => {
+			recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_READY);
+			triggerVideoMetric('ready');
+			// only add the events after the player is ready
+			jwPlayerPlaybackTracker({ event_name: 'video_player_ready' });
+			addBaseTrackingEvents(playerInstance);
+
+			if (onReady) {
+				console.debug('Player ready...');
+				onReady(playerInstance);
+			}
+		});
+
+		playerInstance.on(JWEvents.COMPLETE, () => {
+			if (typeof onComplete === 'function') {
+				onComplete();
+			}
+
+			// Incrementing videos watched count
+			videoIndexRef.current += 1;
+
+			// if stopAutoAdvanceOnExitViewport is false then we don't want to stop the auto advance, keep normal behavior
+			if (!stopAutoAdvanceOnExitViewport) {
+				return;
+			}
```

Comment:
Is this code necessary? It seems like it does not do anything, as there is nothing after this condition anyway.

Reply:
Good question. I'll look into it at the end and remove if it's not necessary 👌